vm/chef/cookbooks/elasticsearch/files/elasticsearch-utils

```diff
@@ -94,15 +95,26 @@ function change_user_password() {
 
   local -r user_to_change="$4"
   local -r new_pass="$5"
+  local -r protocol="$6"
 
-  local -r changepass_url="http://${hostname}:9200/_xpack/security/user/${user_to_change}/_password"
+  local -r changepass_url="${protocol}://${hostname}:9200/_security/user/${user_to_change}/_password"
+  local response=""
 
   echo "Changing elasticsearch password"
-  local -r response="$(curl -X POST \
-    -u "${auth_user}:${auth_user_pass}" \
-    -d "{\"password\":\"${new_pass}\"}" \
-    -H "Content-Type: application/json" \
-    "${changepass_url}")"
+
+  if [[ "${protocol}" == "http" ]]; then
+    response="$(curl -X POST \
+      -u "${auth_user}:${auth_user_pass}" \
```

Comment:
better to put curl arguments to a variable to avoid code duplication

Reply:
Done.